extras/edit_thresholds/editThresholds.R

```diff
@@ -1,155 +1,159 @@
-      df3 <- df3%>%filter(conceptId==row$conceptId &
-                          unitConceptId==row$unitConceptId)}
-    return(df3$index)
-}
-
-# Edit threshold
-edit_threshold <- function(df, index, row){
-  
-  # threshold
-  df[index, "Threshold"] = row$Threshold
-  
-  # notes (avoid duplication)
-  notes_old <- as.character(df[index,'Notes'])
-  if(notes_old=="character(0)"){notes_old=""}
-  notes_old <- stringr::str_replace_all(string=notes_old, row$Notes, "")
-  notes_old <- stringr::str_replace_all(notes_old, "[|][|]", "")
-  notes_new <- paste(notes_old, row$Notes, sep="|")
-  df[index, 'Notes'] = notes_new
-  
-  return(df)
+    return(df_wide)
+  }
   
-}
-
-main_function <- function(file, file_new, thresholds){
-  # 1) get longer table
-  df <- read.csv(file, colClasses = "character", stringsAsFactors = FALSE)
-  df_long <- pivot_longer_func(df)
   
-  # 2) Edit thresholds
-  for(i in c(1:nrow(thresholds))){
-    row = thresholds[i,]
-    index <- get_threshold_location(df_long, row)
-    df_long <- edit_threshold(df_long, index, row)
+  # Get index of rows that fulfill the conditions given in 'file'
+  # If working well, only one row in 'df' is subset for each row in 'file'
+  # Prints the corresponding index.
+  get_threshold_location <- function(df, row){
+    
+      df2 <- df %>% mutate(index=c(1:nrow(.)))
+      
+      # for all
+      df3 <- df2 %>%
+        filter(
+          checkName == row$checkName &
+          cdmTableName == row$cdmTableName)
+      
+      # for Field+Concept Level only
+      if(tolower(row$Level) %in% c("field", "concept")){
+        df3<-df3%>%filter(cdmFieldName == row$cdmFieldName)
+      }
+      
+      # 'additional' auxiliary columns
+      if(row$checkName=="isForeignKey"){
+        df3<-df3%>%filter(fkTableName==row$fkTableName)}
+      if(row$checkName=="plausibleGender"){
+        df3<-df3%>%filter(conceptId==row$conceptId)}
+      if(tolower(row$Level) == "concept" & 
+        row$checkName%in%c("plausibleValueLow", "plausibleValueHigh")){
+        df3 <- df3%>%filter(conceptId==row$conceptId &
+                            unitConceptId==row$unitConceptId)}
+      return(df3$index)
   }
   
-  # 3) get originally wide table
-  df_new <- pivot_wider_func(df=df_long)
-  df_new <- df_new[colnames(df)]
+  # Edit threshold
+  edit_threshold <- function(df, index, row){
+    
+    # threshold
+    df[index, "Threshold"] = row$Threshold
+    
+    # notes (avoid duplication)
+    notes_old <- as.character(df[index,'Notes'])
+    if(notes_old=="character(0)"){notes_old=""}
+    notes_old <- stringr::str_replace_all(string=notes_old, row$Notes, "")
+    notes_old <- stringr::str_replace_all(notes_old, "[|][|]", "")
+    notes_new <- paste(notes_old, row$Notes, sep="|")
+    df[index, 'Notes'] = notes_new
+    
+    return(df)
+    
+  }
   
-  # 4) save it!
-  write.csv(df_new, file_new, row.names = FALSE)
+  # Edits one file
+  main_function <- function(file, thresholds){
```

Comment:
A more descriptive function name would help as this is the function that people will use. e.g. `editThresholds`?

Also, if I am correct, the other functions are helper functions for this one. That is fine, I think the convention is to prepend them with a dot to indicate they are only used internally (e.g. `.pivot_longer_func`).

Reply:
Yep, done! 
I had all hidden functions inside one big function, but it makes more sense this way